src/scenechange/mod.rs

```diff
@@ -0,0 +1,76 @@
+use encoder::Frame;
+use api::FrameInfo;
+
+/// Detects fast cuts using changes in colour and intensity between frames.
+/// Since the difference between frames is used, only fast cuts are detected
+/// with this method. This is probably fine for the purpose of choosing keyframes.
+pub struct SceneChangeDetector {
+  /// Minimum difference between YUV deltas that will trigger a scene change.
+  threshold: u8,
+  /// Frame number and YUV values of the last frame analyzed
+  last_frame: Option<(usize, Vec<(u8, u8, u8)>)>,
+  input_bit_depth: u8,
+}
+
+impl Default for SceneChangeDetector {
+  fn default() -> Self {
+    Self {
+      threshold: 12,
```

Comment:
How did you determine the threshold?

Reply:
Experimentation and comparison with the original implementation I was copying from at https://pyscenedetect.readthedocs.io/en/latest/reference/detection-methods/. The Python implementation uses HSV values and a threshold of 30. I found that comparing the YUV values was sufficient in the cases I tested, and avoided a more costly YUV->RGB->HSV conversion, but that the deltas needed to be scaled down. The deltas for keyframes in YUV were about 1/3 to 1/2 of what they were in HSV, but non-keyframes were very unlikely to have a delta greater than 3 in YUV, whereas they may reach into the double digits in HSV.